PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelperData.h

```diff
@@ -0,0 +1,43 @@
+#ifndef ALIANALYSISTASKEMCALEMBEDDINGHELPERDATA_H
+#define ALIANALYSISTASKEMCALEMBEDDINGHELPERDATA_H
+
+#include <vector>
+
+#include "AliAnalysisTaskEmcalEmbeddingHelper.h"
+#include "THistManager.h"
+
+/**
+ * \file AliAnalysisTaskEmcalEmbeddingHelperData.h
+ * \brief Declaration of class AliAnalysisTaskEmcalEmbeddingHelperData
+ *
+ * Class inheriting from AliAnalysisTaskEmcalEmbeddingHelper which adds 
+ * more functionality to embedding helper, specifically to be used
+ * when embedding data to select events with tracks in a given pT range
+ */
+
+class AliAnalysisTaskEmcalEmbeddingHelperData : public AliAnalysisTaskEmcalEmbeddingHelper{
+ public:
+
+  AliAnalysisTaskEmcalEmbeddingHelperData()                          ;
+  AliAnalysisTaskEmcalEmbeddingHelperData(const char *name)          ;
+  virtual ~AliAnalysisTaskEmcalEmbeddingHelperData()                 ;
+
+  void            UserCreateOutputObjects() ;
+
+  void            AddPtSelection(Double_t min, Double_t max) { fPtSelectionMin.push_back(min); fPtSelectionMax.push_back(max);}
```

Comment:
It might also be nice to override `RetrieveTaskPropertiesFromYAMLConfig()` and allow these to be set there. You'll need to make it virtual